docs/user-event/api-utility.mdx

```diff
@@ -0,0 +1,179 @@
+---
+id: utility
+title: Utility APIs
+---
+
+The following APIs don't have one-to-one equivalents in a real user
+interaction.  
+Their behavior is therefore an interpretation how the "perceived" user
+interaction might be translated to actual events on the DOM.
+
+## clear()
+
+```ts
+clear(element: Element): Promise<void>
+```
+
+This API can be used to easily clear an editable element.
+
+1. Focus element
+1. Select all contents as per browser menu
+1. Delete contents as per browser menu
+
+```jsx
+test('clear', async () => {
+  render(<textarea value="Hello, World!" />)
+
+  await userEvent.clear(screen.getByRole('textbox'))
+
+  expect(screen.getByRole('textbox')).toHaveAttribute('value', '')
+})
+```
+
+The `Promise` is rejected if the element can not be focused or contents can not
+be selected.
+
+## <a name="selectoptions"/><a name="deselectoptions"/> selectOptions(), deselectOptions()
+
+```ts
+selectOptions(
+    element: Element,
+    values: HTMLElement | HTMLElement[] | string[] | string,
```

Comment:
If you file a PR for this, it could as well accept a RegExp.
To be blunt: I don't like the "Utility" APIs at all. I think they should be out of scope because there is no "correct" way to implement them. You can not open a browser and record what happens if you perform the action declared by the API call because the API does not unambiguously declare a user interaction.
But I'm content with separating them as APIs that follow a slightly different approach than the rest of the library.

Reply:
Seems fair to me.